lib/ui/src/components/preview/preview.tsx

```diff
@@ -165,7 +165,10 @@ const Preview = React.memo<PreviewProps>((props) => {
             storyId: id,
             viewMode,
             options: {
-              target: refId ? `storybook-ref-${refId}` : 'storybook-preview-iframe',
+              target:
+                refId && refId !== 'storybook_internal'
+                  ? `storybook-ref-${refId}`
+                  : 'storybook-preview-iframe',
```

Comment:
This targetting code seems to be repeated a bunch of times, maybe we can make `api.emit()` do it internally?